### PR TITLE
Check gifs on CI

### PR DESCRIPTION
**Pull request by romanowski**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -160,6 +160,11 @@ jobs:
     - name: Test cookbooks
       run: .github/scripts/check_docs.sh && .github/scripts/check_examples.sh
 
+    - name: Test gifs
+      run: |
+        SKIP_RENDER=true
+        gifs/generate_gifs.sh $(ls gifs/scenarios/)
```
Comment (romanowski) [inline .github/workflows/ci.yml]:
```suggestion
      run: gifs/generate_gifs.sh $(ls gifs/scenarios/) 
```

File: gifs/run_scenario.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  echo "Please provide one scenario file!"
+  exit 1
+fi
+
+
+fileName=${1##*/}
+name=${fileName%%.sh}
+script=$SCRIPT_DIR/scenarios/$name.sh 
+
+ls $script
```
Comment (romanowski) [inline gifs/run_scenario.sh:17]:
```suggestion
# Make sure that script exisits
test -f $script
```

File: gifs/Dockerfile [hunk 3]
```diff
@@ -21,7 +19,11 @@ RUN apt-get install -y nodejs
...
+# Update path to point to newer version
+RUN curl -fLo scala-cli.deb https://github.com/VirtusLab/scala-cli/releases/download/v0.0.7/scala-cli-x86_64-pc-linux.deb
+RUN dpkg -i scala-cli.deb
+
 COPY *.sh /data/
 COPY scenarios /data/scenarios
 
-ENTRYPOINT scenarios/$1.sh -n && asciinema rec --overwrite --command="scenarios/$1.sh -n" /out/$1.cast
\ No newline at end of file
+ENTRYPOINT ./run_scenario.sh $1
```
Comment (tpasternak) [inline gifs/Dockerfile]: Always good to surround $1 with `"`

File: gifs/generate_gifs.sh [hunk 4]
```diff
@@ -1,23 +1,43 @@
 #!/usr/bin/env bash
 
+set -e
```
Comment (tpasternak) [inline gifs/generate_gifs.sh]: always good to add `set -euxo pipefail`, also in other scripts

File: gifs/generate_gifs.sh [hunk 5]
```diff
@@ -1,23 +1,43 @@
...
 # Generate svg files for arguments based on create scripts with scenarios
 
 SCRIPT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" &> /dev/null && pwd)
 OUT=$SCRIPT_DIR/.scala
-mkdir $OUT
+
+test -d $OUT || mkdir $OUT 
+test -f $OUT/failures.txt && rm $OUT/failures.txt
+
+docker build gifs --tag gif-renderer
```
Comment (tpasternak) [inline gifs/generate_gifs.sh]: `docker build "$SCRIPT_DIR/gifs"`? Same for the next build command.

File: gifs/generate_gifs.sh [hunk 6]
```diff
@@ -1,23 +1,43 @@
...
+
+  # Run the scenario
+  docker run --rm $TTY_OPS -v  $OUT/.scala:/data/out gif-renderer ./run_scenario.sh  $name || (
+    echo "Scenario failed: $name" &&
+    echo $name >> $OUT/failures.txt
+  )
+
+  # do not render gifs without TTY
+  if [ -n "$TTY_OPS" ]; then
+    docker run --rm $svg_render_mappings svg_rendrer a $svg_render_ops --out /data/$name.svg --profile "/profiles/light"
```
Comment (tpasternak) [inline gifs/generate_gifs.sh]: Contrary to the comment above, we generate svgs here, not gifs. (also the filename is generate_gifs.sh)

File: gifs/run_scenario.sh [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+SCRIPT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" &> /dev/null && pwd)
+
+if [ "$#" != "1" ]; then
+  echo "Please provide one scenario file!"
+  exit 1
+fi
+
+
+fileName=${1##*/}
```
Comment (tpasternak) [inline gifs/run_scenario.sh]: imo $(basename "$1) would probably be more self-explainatory

File: gifs/run_scenario.sh [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  # without just run the command 
+  ( 
+    export ASCIINEMA_REC=true &&
+    # remove magic from demo...
+    cp $SCRIPT_DIR/demo-no-magic.sh $SCRIPT_DIR/demo-magic.sh &&
+    $script -n
+  )
+
+test -f status.txt || (
+  echo "Scenarion $sctip failed." &&
```
Comment (tpasternak) [inline gifs/run_scenario.sh]: sctip -> script

File: gifs/run_scenario.sh [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+#warmup
+$script -n
+echo "Done with $?"
+
+test -f status.txt && rm status.txt
+
+#do recording
+tty &&
+  # do recording with tty
+  asciinema rec --overwrite --command="$script -n" $SCRIPT_DIR/out/$name.cast ||
```
Comment (tpasternak) [inline gifs/run_scenario.sh]: i would sarround it in braces, so
(tty && asciinema) || ...

File: gifs/scenarios/debug.sh [hunk 10]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
```
Comment (tpasternak) [inline gifs/scenarios/debug.sh]: What is the purpose of this file?

**Review by tpasternak — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 11]
```diff
@@ -160,6 +160,11 @@ jobs:
     - name: Test cookbooks
       run: .github/scripts/check_docs.sh && .github/scripts/check_examples.sh
 
+    - name: Test gifs
```
Comment (lwronski) [inline .github/workflows/ci.yml:163]: Remove empty line

File: gifs/Dockerfile [hunk 12]
```diff
@@ -21,7 +19,11 @@ RUN apt-get install -y nodejs
 RUN apt-get install -y rubygems
 RUN gem install rouge
 
+# Update path to point to newer version
+RUN curl -fLo scala-cli.deb https://github.com/VirtusLab/scala-cli/releases/download/v0.0.7/scala-cli-x86_64-pc-linux.deb
```
Comment (lwronski) [inline gifs/Dockerfile:23]: I don't think it is good idea to hardcode scala-cli version. Maybe, we should have some configuration file where scala-cli version could be set. 

This file can be updated concurrently with [updating standalone launcher](https://github.com/VirtusLab/scala-cli/blob/5848d09f6919552e7265ed6660f00ae9e7c60ef4/build.sc#L685) in react of release scala-cli.


You can add a hard-core scala-cli version file and then I can update the task in mill.

File: gifs/scenarios/debug.sh [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+  . $SCRIPT_DIR/../demo-magic.sh
+  # hide the evidence
+  clearConsole
+
+  # Put your stuff here
+  # pe "echo 'println(\"<description>\")' | scala-cli -"
+
+  # Wait a bit to read output of last command
+  # doSleep 2
+  echo " " && echo "ok" > status.txt && echo "ok" > status.txt
```
Comment (lwronski) [inline gifs/scenarios/debug.sh]: Duplicated `echo "ok" > status.txt`

File: gifs/generate_gifs.sh [hunk 14]
```diff
@@ -1,23 +1,43 @@
...
 
+set -e
+
 # Generate svg files for arguments based on create scripts with scenarios
 
 SCRIPT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" &> /dev/null && pwd)
 OUT=$SCRIPT_DIR/.scala
-mkdir $OUT
+
+test -d $OUT || mkdir $OUT
```
Comment (lwronski) [inline gifs/generate_gifs.sh]:
```suggestion
mkdir -p "$OUT"
```

File: gifs/generate_gifs.sh [hunk 15]
```diff
@@ -1,23 +1,43 @@
...
+set -e
+
 # Generate svg files for arguments based on create scripts with scenarios
 
 SCRIPT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" &> /dev/null && pwd)
 OUT=$SCRIPT_DIR/.scala
-mkdir $OUT
+
+test -d $OUT || mkdir $OUT 
+test -f $OUT/failures.txt && rm $OUT/failures.txt
```
Comment (lwronski) [inline gifs/generate_gifs.sh]:
```suggestion
rm -f "$OUT/failures.txt"
```

File: gifs/Dockerfile (hunk 12 above)
Comment (romanowski) [reply gifs/Dockerfile:23]: We can do that as a followup

**Pull request merged**